### PR TITLE
Implements GetStats: RTCCodecStats 

### PR DESCRIPTION
**Pull request by obasajujoshua31**
implemented getstats api for rpccodec

#### Reference issue
https://github.com/pion/webrtc/issues/610
Fixes #...

### CODE REVIEW

File: stats_go.go [hunk 1]
```diff
@@ -61,3 +61,18 @@ func (r StatsReport) GetICECandidatePairStats(c *ICECandidatePair) (ICECandidate
 	}
 	return candidateStats, true
 }
+
+// GetCodecStats is a helper method to return the associated stats for a given Codec
+func (r StatsReport) GetCodecStats(c *RTPCodec) (CodecStats, bool) {
```
Comment (Sean-Der) [inline stats_go.go:66]: `GetMediaStats` is probably better. I need to look at `pc-stats` and see if a better name comes out.

**Comment by Sean-Der** [comment]
Look pretty good @obasajujoshua31 I am still reviewing but would you mind changing the commit message to something like this.

```
Add GetMediaStats to StatsReport

Add an API to get basic stats around media. 

Relates to #610 
```

If I am reading `git log` the current commit message makes it really hard to understand what is going on. thanks!

File: mediaengine.go [hunk 2]
```diff
@@ -411,3 +414,21 @@ type RTPCapabilities struct {
...
 }
+
+func (r *RTPCodec) collectStats(collector *statsReportCollector, codecType CodecType) {
+	collector.Collecting()
+
+	stats := CodecStats{
+		Timestamp:   statsTimestampFrom(time.Now()),
+		Type:        StatsTypeCodec,
+		ID:          "codec",
+		PayloadType: uint32(r.PayloadType),
```
Comment (tarrencev) [inline mediaengine.go:425]: This type can be modified to `uint8` in the `CodecStats` struct definition

File: mediaengine.go [hunk 3]
```diff
@@ -411,3 +414,21 @@ type RTPCapabilities struct {
...
+		MimeType:    r.MimeType,
+		ClockRate:   r.ClockRate,
+		Channels:    uint32(r.Channels),
```
Comment (tarrencev) [inline mediaengine.go:428]: This is defined as a `unsigned short` in the spec but we can use `uint8` in go. Please update this in the `CodecStats` struct too.

File: mediaengine.go [hunk 4]
```diff
@@ -411,3 +414,21 @@ type RTPCapabilities struct {
...
+		SDPFmtpLine: r.SDPFmtpLine,
+		CodecType:   codecType,
```
Comment (tarrencev) [inline mediaengine.go:430]: > "encode" or "decode", depending on whether this object represents a media format that the implementation is prepared to encode or decode.

@Sean-Der does this make sense in the context of pion?

File: mediaengine.go [hunk 5]
```diff
@@ -411,3 +414,21 @@ type RTPCapabilities struct {
 	Codecs           []RTPCodecCapability
 	HeaderExtensions []RTPHeaderExtensionCapability
 }
+
+func (r *RTPCodec) collectStats(collector *statsReportCollector, codecType CodecType) {
```
Comment (tarrencev) [inline mediaengine.go:418]: `codecType` shouldn't be passed in here

**Review by tarrencev — CHANGES_REQUESTED** [review]

File: peerconnection.go [hunk 6]
```diff
@@ -1950,6 +1950,34 @@ func (pc *PeerConnection) GetStats() StatsReport {
 	pc.mu.Unlock()
 
 	statsCollector.Collect(stats.ID, stats)
+
+	if pc.api.mediaEngine != nil {
+		for _, sender := range pc.GetSenders() {
```
Comment (tarrencev) [inline peerconnection.go:1955]: You can use `GetTransceivers` to get both senders and receivers

File: peerconnection.go [hunk 7]
```diff
@@ -1950,6 +1950,34 @@ func (pc *PeerConnection) GetStats() StatsReport {
 	pc.mu.Unlock()
 
 	statsCollector.Collect(stats.ID, stats)
+
+	if pc.api.mediaEngine != nil {
```
Comment (tarrencev) [inline peerconnection.go:1954]: I dont think this can be `nil`

File: peerconnection.go (hunk 6 above)
Comment (tarrencev) [reply peerconnection.go:1955]: Actually, I dont think this should be derived from the transceivers. It should just be generated based on the current `MediaEngine`. You can define a method `MediaEngine.collectStats` that aggregates codec stats then is called here with `pc.api.mediaEngine.collectStats(...)`

**Pull request merged**